core-web/libs/edit-content/src/lib/fields/dot-edit-content-file-field/dot-edit-content-file-field.component.ts

```diff
@@ -234,14 +233,9 @@ export class DotEditContentFileFieldComponent implements ControlValueAccessor, O
             }
         });
 
-        this.#dialogRef.onClose
-            .pipe(
-                filter((file) => !!file),
-                takeUntilDestroyed()
-            )
-            .subscribe((file) => {
-                this.store.setPreviewFile(file);
-            });
+        this.#dialogRef.onClose.pipe(filter((file) => !!file)).subscribe((file) => {
```

Comment:
You can do `takeUntilDestroyed(this.destroyRef)`

```ts
private readonly destroyRef = inject(DestroyRef);
```

```suggestion
        this.#dialogRef.onClose.pipe(filter((file) => !!file), takeUntilDestroyed(this.destroyRef)).subscribe((file) => {
```

Reply:
PrimeNG complete this observable 

https://github.com/primefaces/primeng/blob/d6c7ffcfc5cb9a8ed355399ddae276c279836ccf/src/app/components/dynamicdialog/dynamicdialog-ref.ts#L13

Reply:
That only works if the close is called. If the component is destroyed without the close being called, you still have the subscription

Reply:
Oh, we handle that here:

https://github.com/dotCMS/core/blob/19d22021f0e7f752fc82965569b60469e6ba7586/core-web/libs/edit-content/src/lib/fields/dot-edit-content-file-field/dot-edit-content-file-field.component.ts#L248-L252

All good!